### DISCUSSION TITLE
dont-replace-your-view-components-with-razor-components/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#34] [Announcements]
# dont-replace-your-view-components-with-razor-components/

In this post I take a brief look at Razor Components, and whether you should consider using their static render mode instead of View Components.

https://andrewlock.net/dont-replace-your-view-components-with-razor-components/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 17 2020, 01:01, in reply to [James Hickey](https://disqus.com/by/jamesmhickey/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/34#discussioncomment-6058207)</em>

---

<p>Don't get me wrong, I think Blazor is super-interesting, I just can't really see a point for the "static" rendering mode! This was the only scenario I could think of, and that doesn't seem to be worth the effort :) It feels like it's only there because it was easy to implement!</p><p>If you're going to embed the full "Blazor" interactivity into a Razor Page, that's a different story. I can absolutely see the benefits there. But in that case you're pushing an isolated pocket of interactivity into the Razor Page. That differs from this post where I was trying to use Razor Components _within_ the overall Razor Pages paradigm. That's the part that doesn't seem worth it :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 17 2020, 01:51, in reply to [Matt](https://disqus.com/by/disqus_Kf5j1pYhCT/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/34#discussioncomment-6058213)</em>

---

<p>Thanks! This one right <a href="https://ml-software.ch/posts/using-razor-components-in-razor-pages-or-mv-views?" rel="nofollow noopener" title="https://ml-software.ch/posts/using-razor-components-in-razor-pages-or-mv-views?">https://ml-software.ch/posts/using-razor-components-in-razor-pages-or-mv-views?</a> Nice to hear you've come to the same conclusion now - I was in the same boat as you initially, it's only after going a bit deeper that the issues became apparent! 🙂</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 17 2020, 10:54, in reply to [Chris Pratt](https://disqus.com/by/chrisdpratt/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/34#discussioncomment-6058219)</em>

---

<p>Have you been using them with <i>static</i> rendering though? I absolutely agree there's gains there if you're going to buy in to the Blazor model (i.e. a stateful service / webassembly). In this post I was looking at Razor Components as a straight-up replacement for view Components, i.e. stateless, static rendering, no SignalR, no webassembly.</p><p>RE: NavigationManager - not sure how that solves the problem in this static rendering context? Sure, I can use NavigationManager.BaseUri(), but that doesn't seem to buy me a lot - I still need to enter the URL rather than the path to a Razor Page, which I'm doing anyway? Am I missing something there (I could well be, I've only dabbled with Blazor so far 🙂)?</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 23 2020, 01:25, in reply to [Chris Pratt](https://disqus.com/by/chrisdpratt/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/34#discussioncomment-6058223)</em>

---

<p>Interesting, thanks, that pretty much confirms my thoughts :)</p><p>&gt; If you were *only* using static razor components, I guess there's no much point</p><p>That's exactly what I was looking at in this post. As soon as you're using non-static components you're essentially creating a Blazor app, and need to worry about signalr, connectivity etc. Sounds like we actually do agree on that part.</p><p>&gt; If you want to use razor components at all (for behavior), it makes more sense to use static razor components than view components</p><p>Absolutely - I wasn't trying to suggest that view components are ever going to be a good replacement for Razor Components, just that statically-rendered Razor Components aren't a good replacement for view components 🙂</p><p>Thanks for talking it through!</p>

[a comment by an automated account is not shown]

**Comment by lucidmist** [comment]
Thanks for sharing this. Just wondering, would it be challenging to add rest endpoints for this setup to facilitate data sharing across different clients, or would we be better off just using Blazor clean architecture for that purpose?
